TargetTester.cpp

```diff
@@ -445,12 +445,10 @@ void TargetProtoTester::run_test(std::function<void(void)> fun,
 		fun();
 	}
 	catch (ETarget& err) {
-		status = err.status;
+		if(err._status != expected_result)
```

Comment:
This change alters the function's logic. In the case of negative tests, we will not detect that a `fun` function incorrectly succeeded.

Reply:
Exception is re-thrown from catch block and will propagate until another catch block (catching this type of exception).
So technically the final result is the same.

Reply:
I had in mind a situation where it calls a function that should throw an exception. However, due to an error, it doesn't do it and silently accepts the wrong parameters. In this situation, the error code check will not be performed and we will not detect that we expected an exception.

Reply:
I think I don't understand what you mean specifically, but ok I'll just revert that change. The main goal of this PR is just to have a project compiled successfully.

Reply:
Something like this:
```run_test([] { }, Protocol::STATUS_PROTECTED_ADDR);```
It should throw 'Invalid result' exception.

Reply:
> Something like this: `run_test([] { }, Protocol::STATUS_PROTECTED_ADDR);` It should throw 'Invalid result' exception.

And it would.